src/main/java/com/minecolonies/coremod/colony/requestsystem/data/StandardRequestSystemBuildingDataStore.java

```diff
@@ -0,0 +1,212 @@
+{
+
+    private IToken<?>                                id;
+    private final Map<TypeToken<?>, Collection<IToken<?>>> openRequestsByRequestableType;
+    private final Map<Integer, Collection<IToken<?>>>      openRequestsByCitizen;
+    private final Map<Integer, Collection<IToken<?>>>      completedRequestsByCitizen;
+    private final Map<IToken<?>, Integer>                  citizenByOpenRequest;
+
+    public StandardRequestSystemBuildingDataStore(
+      final IToken<?> id,
+      final Map<TypeToken<?>, Collection<IToken<?>>> openRequestsByRequestableType,
+      final Map<Integer, Collection<IToken<?>>> openRequestsByCitizen,
+      final Map<Integer, Collection<IToken<?>>> completedRequestsByCitizen,
+      final Map<IToken<?>, Integer> citizenByOpenRequest) {
+        this.id = id;
+        this.openRequestsByRequestableType = openRequestsByRequestableType;
+        this.openRequestsByCitizen = openRequestsByCitizen;
+        this.completedRequestsByCitizen = completedRequestsByCitizen;
+        this.citizenByOpenRequest = citizenByOpenRequest;
+    }
+
+    public StandardRequestSystemBuildingDataStore()
+    {
+        this(StandardFactoryController.getInstance().getNewInstance(TypeConstants.ITOKEN),
+          new HashMap<>(),
+          new HashMap<>(),
+          new HashMap<>(),
+          new HashMap<>());
+    }
+
+    @Override
+    public Map<TypeToken<?>, Collection<IToken<?>>> getOpenRequestsByRequestableType()
+    {
+        return openRequestsByRequestableType;
+    }
+
+    @Override
+    public Map<Integer, Collection<IToken<?>>> getOpenRequestsByCitizen()
+    {
+        return openRequestsByCitizen;
+    }
+
+    @Override
+    public Map<Integer, Collection<IToken<?>>> getCompletedRequestsByCitizen()
+    {
+        return completedRequestsByCitizen;
+    }
+
+    @Override
+    public Map<IToken<?>, Integer> getCitizensByRequest()
+    {
+        return citizenByOpenRequest;
+    }
+
+    @Override
+    public IToken<?> getId()
+    {
+        return id;
+    }
+
+    @Override
+    public void setId(final IToken<?> id)
+    {
+        this.id = id;
+    }
+
+    public static class Factory implements IFactory<FactoryVoidInput, StandardRequestSystemBuildingDataStore>
+    {
+
+        @NotNull
+        @Override
+        public TypeToken<? extends StandardRequestSystemBuildingDataStore> getFactoryOutputType()
+        {
+            return TypeToken.of(StandardRequestSystemBuildingDataStore.class);
+        }
+
+        @NotNull
+        @Override
+        public TypeToken<? extends FactoryVoidInput> getFactoryInputType()
+        {
+            return TypeConstants.FACTORYVOIDINPUT;
+        }
+
+        @NotNull
+        @Override
+        public StandardRequestSystemBuildingDataStore getNewInstance(
+          @NotNull final IFactoryController factoryController, @NotNull final FactoryVoidInput factoryVoidInput, @NotNull final Object... context) throws IllegalArgumentException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)